js/gateio.js

```diff
@@ -2263,6 +2263,115 @@ module.exports = class gateio extends Exchange {
         };
     }
 
+    parsePosition (position, market = undefined) {
+        // [{
+        //     value: "12.475572",
+        //     leverage: "0",
+        //     mode: "single",
+        //     realised_point: "0",
+        //     contract: "BTC_USDT",
+        //     entry_price: "62422.6",
+        //     mark_price: "62377.86",
+        //     history_point: "0",
+        //     realised_pnl: "-0.00624226",
+        //     close_order:  null,
+        //     size: "2",
+        //     cross_leverage_limit: "25",
+        //     pending_orders: "0",
+        //     adl_ranking: "5",
+        //     maintenance_rate: "0.005",
+        //     unrealised_pnl: "-0.008948",
+        //     user: "6645677",
+        //     leverage_max: "100",
+        //     history_pnl: "14.98868396636",
+        //     risk_limit: "1000000",
+        //     margin: "0.740721495056",
+        //     last_close_pnl: "-0.041996015",
+        //     liq_price: "59058.58"
+        // }]
+        const contract = this.safeValue (position, 'contract');
+        market = this.safeMarket (contract, market);
+        const now = this.milliseconds ();
+        return {
+            'info': position,
+            'symbol': market['symbol'],
+            'timestamp': now,
+            'datetime': this.iso8601 (now),
+            'initialMargin': position['margin'],
```

Comment:
We should not pass all these values as is, all of them should go through safe-methods for reasons covered her:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#accessing-dictionary-keys
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods